src/pragma.rs

```diff
@@ -0,0 +1,376 @@
+//! Pragma helpers
+
+use std::ops::Deref;
+
+use {Connection, DatabaseName, Result, Row};
+use error::Error;
+use ffi;
+use types::{ToSql, ToSqlOutput, ValueRef};
+
+pub struct Sql {
+    buf: String,
+}
+
+impl Sql {
+    pub fn new() -> Sql {
+        Sql { buf: String::new() }
+    }
+
+    pub fn push_pragma(&mut self,
+                       schema_name: Option<DatabaseName>,
+                       pragma_name: &str)
+                       -> Result<()> {
+        try!(self.push_keyword("PRAGMA"));
+        self.push_space();
+        if let Some(schema_name) = schema_name {
+            self.push_schema_name(schema_name);
+            self.push_dot();
+        }
+        self.push_keyword(pragma_name)
+
+    }
+
+    pub fn push_keyword(&mut self, keyword: &str) -> Result<()> {
+        if !keyword.is_empty() && is_identifier(keyword) {
```

Comment:
This is an interesting solution but feels too general to me; I worry we've missed some way of abusing this. What if we exposed a very limited set of (essentially hard-coded) pragmas, either via an enum or functions, and expanded that list as needed over time, instead of allowing arbitrary strings? Especially since users can already run arbitrary strings as queries if they want.